gemini/sample-apps/finance-advisor-spanner/Schema-Operations.sql

```diff
@@ -0,0 +1,152 @@
+UPDATE EU_MutualFunds SET investment_strategy_Embedding_vector = investment_strategy_Embedding WHERE investment_strategy_Embedding is not NULL and  EXTRACT(YEAR from inception_date)  = 1973;
+UPDATE EU_MutualFunds SET investment_strategy_Embedding_vector = investment_strategy_Embedding WHERE investment_strategy_Embedding is not NULL and  EXTRACT(YEAR from inception_date)  = 1981;
+UPDATE EU_MutualFunds SET investment_strategy_Embedding_vector = investment_strategy_Embedding WHERE investment_strategy_Embedding is not NULL and  EXTRACT(YEAR from inception_date)  = 2016;
+UPDATE EU_MutualFunds SET investment_strategy_Embedding_vector = investment_strategy_Embedding WHERE investment_strategy_Embedding is not NULL and  EXTRACT(YEAR from inception_date)  = 2020;
+UPDATE EU_MutualFunds SET investment_strategy_Embedding_vector = investment_strategy_Embedding WHERE investment_strategy_Embedding is not NULL and  EXTRACT(YEAR from inception_date)  = 2000;
+UPDATE EU_MutualFunds SET investment_strategy_Embedding_vector = investment_strategy_Embedding WHERE investment_strategy_Embedding is not NULL and  EXTRACT(YEAR from inception_date)  = 1992;
+
+
+CREATE SEARCH INDEX
+  category_Tokens_IDX
+ON
+  EU_MutualFunds(category_Tokens);
+CREATE SEARCH INDEX
+  fund_benchmark_Tokens_IDX
+ON
+  EU_MutualFunds(fund_benchmark_Tokens);
+CREATE SEARCH INDEX
+  fund_name_Tokens_IDX
+ON
+  EU_MutualFunds(fund_name_Tokens);
+CREATE SEARCH INDEX
+  investment_managers_Tokens_IDX
+ON
+  EU_MutualFunds(investment_managers_Tokens);
+CREATE SEARCH INDEX
+  investment_strategy_Tokens_IDX
+ON
+  EU_MutualFunds(investment_strategy_Tokens);
+CREATE SEARCH INDEX
+  morningstar_benchmark_Tokens_IDX
+ON
+  EU_MutualFunds(morningstar_benchmark_Tokens);
+CREATE SEARCH INDEX
+  top5_holdings_Tokens_IDX
+ON
+  EU_MutualFunds(top5_holdings_Tokens);
+CREATE SEARCH INDEX
+  top5_regions_Tokens_IDX
+ON
+  EU_MutualFunds(top5_regions_Tokens);
+CREATE SEARCH INDEX
+  investment_managers_Substring_Tokens_IDX
+ON
+  EU_MutualFunds(investment_managers_Substring_Tokens);
+CREATE SEARCH INDEX
+  investment_managers_Substring_investment_Strategy_Tokens_Combo_IDX
+ON
+  EU_MutualFunds(investment_managers_Substring_Tokens,
+    investment_strategy_Tokens);
+
+
+CREATE SEARCH INDEX
+  investment_managers_Substring_NgRAM_investment_Strategy_Tokens_Combo_IDX
+ON
+  EU_MutualFunds(investment_strategy_Tokens,
+    investment_managers_Substring_Tokens_NGRAM);
+CREATE VECTOR INDEX
+  InvestmentStrategyEmbeddingIndex
+ON
+  EU_MutualFunds(investment_strategy_Embedding_vector)
+WHERE
+  investment_strategy_Embedding_vector IS NOT NULL OPTIONS ( tree_depth = 2,
+    num_leaves = 40,
+    distance_type = 'EUCLIDEAN' );
+
+CREATE SEARCH INDEX
+  investment_managers_Substring_Tokens_with_vectors_NGRAM_IDX
+ON
+  EU_MutualFunds(investment_managers_Substring_Tokens_NGRAM) STORING (investment_strategy_Embedding_vector);
+
+
+
```

Comment:
Please fix this spacing throughout the file.

Reply:
done